src/utils/network-monitor.ts

```diff
@@ -2180,12 +2178,17 @@ export class NetworkMonitor {
         canFail: waitForReceipt ? false : canFail, // canFail,
       })
       if (receipt === null) {
+        if (!waitForReceipt) {
```

Comment:
Can you add a note why wallet nonce is updated only if not waitForReceipt

Reply:
If we're not waiting for receipt and receipt is null, then we can assume that the transaction was submitted and will be mined eventually. If there is no need for a receipt, we can immediately increment the wallet nonce and just move forward. But if there is a need for receipt but receipt is null, this indicates that something may have happened that is unexpected, we do not increment the nonce and let the function calling logic to step in and do a review before incrementing the nonce. Like this there is a way for re-submission of a TX that would overwrite a maybe still pending TX. Will be thinking this through more extensively before settling on a final version to use.